homeassistant/components/cover/reproduce_state.py

```diff
@@ -74,17 +77,32 @@ async def _async_reproduce_state(
         == state.attributes.get(ATTR_CURRENT_POSITION)
     ):
         # Open/Close
+        supported_features = try_parse_enum(
+            CoverEntityFeature,
+            cur_state.attributes.get(ATTR_SUPPORTED_FEATURES),
+        ) or CoverEntityFeature(0)
         if state.state in [CoverState.CLOSED, CoverState.CLOSING]:
-            service = SERVICE_CLOSE_COVER
+            if (
+                CoverEntityFeature.CLOSE in supported_features
+                or CoverEntityFeature.CLOSE_TILT not in supported_features
+            ):
+                service = SERVICE_CLOSE_COVER
+            elif CoverEntityFeature.CLOSE_TILT in supported_features:
+                service = SERVICE_CLOSE_COVER_TILT
```

Comment:
there are covers that support both up/down and tilt, right? I think this won't work for those

Also, on L74 we check for the `ATTR_CURRENT_POSITION` but not `ATTR_CURRENT_TILT_POSITION`

Reply:
> there are covers that support both up/down and tilt, right? I think this won't work for those

Covers only support one tilt https://developers.home-assistant.io/docs/core/entity/cover/?_highlight=cover#properties.  I'm not following how this could be an issue

> 
> Also, on L74 we check for the `ATTR_CURRENT_POSITION` but not `ATTR_CURRENT_TILT_POSITION`

For a tilt only cover this value will likely be empty but I guess there is no guarantee it won't be set

Reply:
A cover can have both `CoverEntityFeature.CLOSE` and `CoverEntityFeature.CLOSE_TILT`. They are not mutually exclusive. So for a cover that supports both "slide" and tilt, this will only reproduce the "slide" part. Am I reading it wrong?

Reply:
I spent another hour or so digging into this and there are so many combinations that we don't handle correctly because we don't look at supported features.

I've come to the conclusion the whole thing needs to be rewritten to check supported features.